### ISSUE TITLE
Require transactions to refer to recent blocks

### ISSUE THREAD

**Issue by bobbinth** [0xPolygonMiden/miden-node#529]
Currently, we have no restriction on how old a block referenced by a transaction needs to be (it could be a genesis block, for example). It may be beneficial to make this a bit more restrictive (and this is done in some other chains - e.g., Solana). For example, we could require that a transaction references a block which is within 256 blocks of the chain tip. This would be a global setting applicable to all transactions. Individual transactions would still be able to reduce this window via individual expiration settings as they can now.

**Comment by igamigo** [comment]
Is there any specific reason why we would not want to initialize the expiration delta to 256 instead of `u32::MAX` on the kernel prologue?

**Comment by bobbinth** [comment]
Good point! Though, we'd set the block expiration time to `block_ref + 256` rather than the delta directly. I'll transfer this issue to `miden-base`.
